src/huggingface_hub/_commit_api.py

```diff
@@ -386,16 +386,40 @@ def _upload_lfs_object(operation: CommitOperationAdd, lfs_batch_action: dict, to
         return
     upload_action = lfs_batch_action["actions"].get("upload")
     verify_action = lfs_batch_action["actions"].get("verify")
-    with operation.as_file(with_tqdm=True) as fileobj:
-        logger.debug(f"Uploading {operation.path_in_repo} as LFS file...")
-        lfs_upload(
-            fileobj=fileobj,
-            upload_action=upload_action,
-            verify_action=verify_action,
-            upload_info=upload_info,
-            token=token,
-        )
-        logger.debug(f"{operation.path_in_repo}: Upload successful")
+    if HF_HUB_ENABLE_HF_TRANSFER:
+        if not isinstance(operation.path_or_fileobj, str):
+            raise ValueError("When using hf_transfer, you should not open the file beforehand and pass the file path as a string instead")
```

Comment:
Uploading files from memory should be faster than from disk. It may be interesting to actually support that, I'll think of something, but maybe in a subsequent PR.
On big machines, network bandwidth > disk read speed (especially for hard drives, SSD not sure) 😄
I'll rewrite this bit so that it defaults to normal upload with a warning (smthing like "hf transfer is enabled but doesn't support uploading files from memory, please pass a file path")

Reply:
Yes, :+1: for warning + default to classic upload and leave it for a subsequent PR

Reply:
Done !